bundles/framework/layerselector2/Flyout.js

```diff
@@ -117,15 +122,20 @@ Oskari.clazz.define('Oskari.mapframework.bundle.layerselector2.Flyout',
          * @method  @private addDefaultFilters
          */
         addDefaultFilters: function() {
-            var me = this;
-
-            // Add newest filter
-            me.addNewestFilter();
-
-            // Add featuredata filter
-            me.addFeaturedataFilter();
+            this.addNewestFilter();
+        },
+        /**
+         * Adds extra filter buttons if layers exists
+         * @method  @private displayFilters
+         */
+        displayExtraFilters() {
+            if ( this.mapLayerService.getFilteredLayers('featuredata').length !== 0) {
```

Comment:
You shouldn't hardcode the filters here. Instead have a function in maplayerservice that returns the "active" filters (returns only the ones that actually hit layers when filtering). It should loop through all the registered filters instead of hardcoded names as filters can be added externally. Layerselector should call the function again and re-render the filters on MapLayerEvent with operation add/remove as they might change the active filters (more layers have been registered to the system or removed).